apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/TransformerUtils.java

```diff
@@ -904,6 +904,25 @@ static Extension createExtensionDate(
     return extension;
   }
 
+  /**
+   * @param ccwVariable the {@link CcwCodebookInterface} being mapped
+   * @param date the value to use for {@link Coding#getCode()} for the resulting {@link Coding}
+   * @return the output {@link Extension}, with {@link Extension#getValue()} set to represent the
+   *     specified input values
+   */
+  static Extension createExtensionDate(CcwCodebookInterface ccwVariable, LocalDate date) {
+    Extension extension = null;
+    try {
+      String stringDate = date.toString();
+      DateType dateValue = new DateType(stringDate);
+      String extensionUrl = CCWUtils.calculateVariableReferenceUrl(ccwVariable);
+      extension = new Extension(extensionUrl, dateValue);
+    } catch (DataFormatException e) {
+      throw new InvalidRifValueException(
+          String.format("Unable to create DateType with date: '%s'.", date), e);
+    }
```

Comment:
What input could cause the DataFormatException here? Not sure that code is reachable given the input, unless I'm missing something.

Also if date is input as null, this will throw a nullpointer due to line 916. We could assertNotNull on date if its never expected to be null here (which i think is the case, given upstream somewhere we use an optional when parsing this field from the database?)

Reply:
Checking if date is null now. DataFormatException should be thrown if new DateType cannot create a dateType.

Reply:
date.toString() will throw a nullPointer if date is null; it wont get to try and parse it.